ironfish/src/network/peerNetwork.ts

```diff
@@ -718,11 +720,73 @@ export class PeerNetwork {
     return new PooledTransactionsResponse(transactions, rpcId)
   }
 
-  private onGetBlockTransactionsRequest(
+  private async onGetBlockTransactionsRequest(
+    peer: Peer,
     message: GetBlockTransactionsRequest,
-  ): GetBlockTransactionsResponse {
-    // TODO(IRO-2279): Implement a handler for this
-    return new GetBlockTransactionsResponse(message.blockHash, [], message.rpcId)
+  ): Promise<GetBlockTransactionsResponse> {
+    const block = await this.chain.db.withTransaction(null, async (tx) => {
+      const header = await this.chain.getHeader(message.blockHash, tx)
+
+      if (header === null) {
+        throw new CannotSatisfyRequestError(
+          `Peer requested transactions for block ${message.blockHash.toString(
+            'hex',
+          )} that isn't in the database`,
+        )
+      }
+
+      if (header.sequence < this.chain.head.sequence - MAX_GET_BLOCK_TRANSACTIONS_DEPTH) {
+        throw new CannotSatisfyRequestError(
+          `Peer requested transactions for block ${message.blockHash.toString(
+            'hex',
+          )} with sequence ${header.sequence} while chain head is at sequence ${
+            this.chain.head.sequence
+          }`,
+        )
+      }
+
+      const block = await this.chain.getBlock(header, tx)
```

Comment:
Any reason we can't just call `getBlock` directly with the hash? Is it because we want to check the sequence first before we query the whole block?

Reply:
Yep -- The block body is relatively hefty compared to the header, so it's worth doing validation on the header first before fetching the rest of the block. It ends up being the same number of database reads, but split up into separate fetches for the header and body.

It might not have been clear, but calling `getBlock` with the hash fetches both the header and the body, and calling `getBlock` with the header fetches only the body.

Reply:
👍🏻